### PR TITLE
feat(docx): Improve search experience in new docs site

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

Searching the new docs site should be smooth and consumer grade! This cleans up some UI cruft and makes the search feel more intuitive.

### Before

- No search icon or clearable on the search input
- Modal would overflow the screen, forcing scroll - also limited results per viewport with a "base" width modal
- Results headings too similar in weight to section headings (Design + Components)
- Could not predictably fire the Search modal via keyboard

https://github.com/user-attachments/assets/cf322dd0-9bfc-40ce-87e7-ad9abe3ebca3

### After

- Search icon on the search input, and it's clearable
- Modal cannot overflow the screen
- Modal is wider, allowing for more results per query
- Better hierarchy within search results (more spacing, emphasis on content over headings)
- Can predictably fire the search modal using keyboard

https://github.com/user-attachments/assets/80b596e6-b0d0-42ac-bf72-2a04574db442

### Added

- 

### Changed

- add clear and icon to input
- use large modal
- style results box
- Nav search item is a Button now

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by chris-at-jobber** [comment]
Notes for additional improvements that might need more dev assistance:

- interaction to fire search by keyboard is janky, cannot open by keyboard interactions (onClick only)
- `useInView` to add gradient overflow indicator when scrolling within modal

**Comment by chris-at-jobber** [comment]
Component usage thoughts:

- If `Box` or utils are desired as replacements CSS modules, should support 
  - max-height
  - overflow
  - box-sizing

- `Heading` should support `subdued` variation for web (parity with mobile)

File: packages/site/src/layout/SearchBox.module.css
Comment (chris-at-jobber) [inline packages/site/src/layout/SearchBox.module.css:1]: See my comment [here](https://github.com/GetJobber/atlantis/pull/2100#issuecomment-2453119640) on why Box could/should support some of these properties

File: packages/site/src/layout/SearchBox.tsx [hunk 1]
```diff
@@ -56,18 +57,26 @@ export const SearchBox = ({
...
+      <Content spacing={"larger"}>
         <InputText
           ref={ref}
           value={search}
           placeholder="Search"
+          prefix={{ icon: "search" }}
+          clearable="always"
           onChange={d => setSearch(d as string)}
         />
-        <Box width="100%">
```
Comment (chris-at-jobber) [inline packages/site/src/layout/SearchBox.tsx:67]: See **[comment](https://github.com/GetJobber/atlantis/pull/2100#issuecomment-2453119640)** on why Box didn't meet the needs here

File: packages/site/src/layout/SearchBox.module.css
Comment (scotttjob) [reply packages/site/src/layout/SearchBox.module.css:1]: How much effort to get Box to support them in your estimate? Is it something I could roll inside an hour or so?

**Review by scotttjob — APPROVED** [review]
LGTM!

Apologies on the delay. I need to clean up my GH notifications, it's 95% noise so I miss some of the signals.

**Pull request merged**